hyperspy/_signals/spectrum.py

```diff
@@ -84,7 +84,10 @@ def _spikes_diagnosis(self, signal_mask=None,
             dc = dc[~navigation_mask, :]
         der = np.abs(np.diff(dc, 1, -1))
         plt.figure()
-        plt.hist(np.ravel(der.max(-1)), 100)
+        if len(dc.shape) == 1:
```

Comment:
Thinking about this a bit more, most spikes will show as low count 
outliers (rarely anything about 5 at the current number of bins) at the 
right of the histogram. The problem is to set the default y range so 
that the user doesn't need to zoom in to see them and estimate the 
suitable threshold. So, what about a simple solution like setting the 
top y limit to something like 30 if the highest bin is above 30?

On 30/07/15 17:00, Tomas Ostasevicius wrote:
>
> In hyperspy/_signals/spectrum.py 
> <https://github.com/hyperspy/hyperspy/pull/626#discussion_r35887181>:
>
> > @@ -84,7 +84,10 @@ def _spikes_diagnosis(self, signal_mask=None,
> >              dc = dc[~navigation_mask, :]
> >          der = np.abs(np.diff(dc, 1, -1))
> >          plt.figure()
> > -        plt.hist(np.ravel(der.max(-1)), 100)
> > +        if len(dc.shape) == 1:
>
> I think by just changing the |ylim| to -1 it should be good enough. 
> Remember this is in log scale still, so leaving it as 0 just cut off 
> any bins with just one count (you can /barely/ see them in the figures 
> you attached... in some cases at least).
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/hyperspy/hyperspy/pull/626/files#r35887181>.
>